addons/event_sale/tests/test_event_sale.py

```diff
@@ -121,3 +122,21 @@ def test_01_ticket_price_with_pricelist_and_tax(self):
         })
         sol.product_id_change()
         self.assertEqual(so.amount_total, 660.0, "Ticket is $1000 but the event product is on a pricelist 10 -> 6. So, $600 + a 10% tax.")
+
+    def test_delete_associated_sale_order(self):
+        """ This test ensures that when deleting a sale order, if the latter is linked to an event registration,
+        the number of expected seats will be correctly updated """
+        event = self.register_person.event_registration_ids.event_id
+        self.register_person.action_make_registration()
+        self.assertEqual(event.seats_expected, 1)
+        self.sale_order.unlink()
+        self.assertEqual(event.seats_expected, 0)
+
+    def test_delete_associated_sale_order_line(self):
```

Comment:
Could be great to run this test as a real user, using the ``@users`` decorator and a sales person